### DISCUSSION TITLE
will there be any support for the roccat pyro in the near future?

### DISCUSSION THREAD

**Discussion by essbo** [X3n0m0rph59/eruption#166] [Hardware Support]
i just recently got a pyro.

sadly i've had to find out that the lights aren't working with my linux machine.

is there anything planned to implement the pyro  ?

ty for your work.

appreciate it
essbo.

**Comment by X3n0m0rph59** [comment]
Hi @essbo, thanks for posting your request!

I currently do not have access to a `ROCCAT Pyro` keyboard. Adding support for that device would require a capture of the init sequence e.g. done with Wireshark. Please find the instructions on how to do this in the [Eruption Wiki](https://github.com/X3n0m0rph59/eruption/wiki), if you are willing to assist.

Please open a GitHub issue using the "Request for Hardware Support" template.

**Comment by Galouve** [comment]
Hello @X3n0m0rph59 , i have a Roccat Pyro, and i had a look into the wiki. i have a windows 10 VM (had to, kb firmware update), Wireshark installed on host side, but i don't understand how to do the tricks with. I'm not used to wireshark , so i'm a bit lost in configuring the stuff. I really want to contribute, your software is amazing, as a Linux user for years. When i connect usbmon, i don't know which one i have to take, so i can try doing something with all of my usb but i don't know if it's okay.
How can i see the right usbmon, or taking the "all-in-one" usbmon0 is okay ? 
Thank you for all your work on this !

Galouve, a French wolf-penguin thing

**Comment by X3n0m0rph59** [comment]
Hi @Galouve, I still do not have access to a `ROCCAT Pyro` keyboard at the moment, so I am unable to test and even to create bus-traffic captures in the first place. So it would be really great if you could provide Wireshark packet capture files... I just edited the Eruption wiki in order to clarify some steps required to create the *.pcapng files. Please feel free to ask if you require further assistance!

**Reply by Galouve** [reply]
Thx for your answer !
I will try to do so in the next days, i'll read the wiki again and i'll send the files to you, so you will see if something is wrong and have to be reworked. 
I'll keep contact if needed.
Have a good day !

Galouve

**Reply by Galouve** [reply]
Hi @X3n0m0rph59,
I'm a bit late, but i think i finally managed to capture sequences for the Roccat Pyro, on w10 VM.
How do i send you the pcapng file, so you could see if all went as expected ?

**Reply by X3n0m0rph59** [reply]
@Galouve Great, thank you very much for your support!

You can simply attach the file by creating a new comment in this issue. (Right below the text input box sits an area where files can be dropped)

**Reply by Galouve** [reply]
It doesn't want to, says it doesn't support this file type :(

**Reply by X3n0m0rph59** [reply]
@Galouve Ah right, please try to compress the file, a *.zip file will definitely work!

**Reply by Galouve** [reply]
[ROCCAT_PYRO.zip](https://github.com/eruption-project/eruption/files/11651442/ROCCAT_PYRO.zip)

Noice, it works !
Hope it helps you well in your dev-work ! :+1:

**Reply by X3n0m0rph59** [reply]
@Galouve Thanks for uploading the capture file!

Seems to have enough data about the protocol and especially RGB packets in it... I will try to implement a driver using this information

**Reply by Galouve** [reply]
Nice to read this, i'm happy knowning it can serve you ! If you need more data, don't hesitate to ask me back, so i can capture this for you !
